### PR TITLE
Enable Loss Fn in Graph PP

### PR DESCRIPTION
**Pull request by sanketpurandare**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #247

### CODE REVIEW

File: autoparallel/api.py [hunk 1]
```diff
@@ -326,6 +348,7 @@ def build_model_graph(self):
                 print_output=False, include_stride=True, include_device=True
             ),
         )
+        print(gm.graph)
```
Comment (xmfan) [inline autoparallel/api.py]: :( use tlparse to get your graphs

**Review by wconstab — APPROVED** [review]

File: autoparallel/api.py [hunk 2]
```diff
@@ -159,17 +159,35 @@ def enable_local_map_wrapping():
         yield
 
 
-def _export(model: torch.nn.Module, inputs: tuple[Any]) -> torch.nn.Module:
+def _export(
+    model: torch.nn.Module, model_wrapper: Callable, inputs: tuple[Any, ...]
```
Comment (wconstab) [inline autoparallel/api.py]: nit: model_wrapper can be optional? if not provided, just use the model itself as the tracing entrypoint?

Comment (sanketpurandare) [reply autoparallel/api.py]: The model wrapper is never None or Optional. Even if we don't use loss function we do construct a model wrapper. I feel this would be a good api even for future use when we decide to use an optimizer.
